core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -89,6 +90,7 @@
   private static final int BLOCK_REPLICATION_CONSTANT = 3;
   /** Lock for initializing the contexts, currently only one set of contexts is supported. */
   private static final Object INIT_LOCK = new Object();
+  private static final String ZOOKEEPER_IDENTIFIER = "zk@";
```

Comment:
not sure if `AbstractFileSystem` is the best place to host these constants

Reply:
also, having this in `AbstractFileSystem.java`, does it mean we only support this `alluxio://zk@` syntax for HDFS compatible interface?

Reply:
Andrew suggested making ```AlluxioURI``` capable of processing all kinds of URI including the alluxio on Zookeeper URI and alluxio on embedded journal URI. AlluxioURI will host all these constants and expose methods as ```getAuthorityType()``` and ```getAuthorityData()```(return zookeeper addresses when authorityType is Zookeeper).

I suggest to put it here now and have a separate PR modify AlluxioURI(probably after gene's improvement or discuss with gene later).

Yes, by now, alluxio://zk@ only useful for hadoop, spark, hbase, flink, and hive(computation frameworks), all those are HDFS compatible interface. Are there any other circumstances we need this syntax?